test/e2e/job_scheduling.go

```diff
@@ -387,4 +392,196 @@ var _ = Describe("Job E2E Test", func() {
 		err = waitJobPhaseReady(context, job)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
```

Comment:
Better add a case with both queue and namespace fair share?

Reply:
This RP is too big now, I'll submit a new test next PR.